assets/src/application/ui/display/icons/svgs/withEnhance.tsx

```diff
@@ -0,0 +1,38 @@
+import React from 'react';
+import classNames from 'classnames';
+import { withSizeProps, IconSize } from './types';
+import './style.scss';
+
+const camelToSnakeCase = (str: string) => {
+	return str
+		.split(/(?=[A-Z])/)
+		.map((x) => x.toLowerCase())
+		.join('-');
+};
+
+const withEnhance = <P extends withSizeProps>(WrappedComponent: React.ComponentType<P>) => {
+	type ISize = { noMargin?: boolean; size: IconSize };
+
+	// @ts-ignore
+	const displayName = WrappedComponent?.render?.name;
+
+	const generatedIconClassName = displayName && 'ee-icon--' + camelToSnakeCase(displayName.replace('Svg', ''));
+
+	const WithSize: React.FC<P & ISize> = ({ noMargin, size, ...props }) => {
+		const className = classNames(
+			'ee-svg',
+			size && `ee__icon--${size}`,
+			noMargin && 'ee__icon--no-margin',
+			generatedIconClassName,
+			props.className
+		);
```

Comment:
HOCception !!!
can this not just be combined with the logic inside `withEnhance` ???

Reply:
splitted, thx